src/porepy/grids/grid_bucket.py

```diff
@@ -508,20 +502,19 @@ def remove_node_props(
         # Do some checks of parameters first
         if g is not None and not isinstance(g, list):
             g = [g]
+        if g is None:
+            g = list(self._nodes)
 
-        keys = list(keys)
+        keys = [keys] if isinstance(keys, str) else list(keys)
 
         for key in keys:
-            if g is None:
-                for _, d in self:
-                    del d[key]
-            else:
-                for h, d in self:
-                    if h in g:
-                        del d[key]
+            for grid in g:
+                data = self._nodes[grid]
+                if key in data:
+                    del data[key]
 
     def remove_edge_props(
-        self, keys: Union[Any, List[Any]], e: List[Tuple[pp.Grid, pp.Grid]] = None
+        self, keys: Union[Any, List[Any]], edges: List[Tuple[pp.Grid, pp.Grid]] = None
```

Comment:
This is really picky, but if you want edges instead of e, should it not also be grids instead of g in the corresponding method for nodes?

Reply:
Generally, I think we should avoid using one-letter variables in the _public_ API as it should be as intuitive as possible.
However this PR wasn't intended as a full overhaul of `GridBucket`, so I haven't by any means been fully consistent with my changes.